src/commonControls/questionnaireControl/QuestionnaireControl.ts

```diff
@@ -0,0 +1,1578 @@
+    private isSpecificAnswerToAskedQuestion(input: ControlInput): boolean {
+        try {
+            okIf(this.state.focusQuestionId !== undefined);
+            const question = this.getQuestionContentById(this.state.focusQuestionId, input);
+
+            okIf(
+                InputUtil.isIntent(
+                    input,
+                    SingleValueControlIntent.intentName(this.props.interactionModel.slotType),
+                ),
+            );
+            const { feedback, action, target, valueStr, valueType } = unpackSingleValueControlIntent(
+                (input.request as IntentRequest).intent,
+            );
+            okIf(InputUtil.targetIsUndefined(target));
+            okIf(InputUtil.valueTypeMatch(valueType, this.getSlotTypes()));
+            okIf(InputUtil.valueStrDefined(valueStr));
+            okIf(InputUtil.feedbackIsMatchOrUndefined(feedback, [$.Feedback.Affirm, $.Feedback.Disaffirm]));
+            okIf(InputUtil.actionIsMatchOrUndefined(action, this.props.interactionModel.actions.answer));
+            return true;
+        } catch (e) {
+            verifyErrorIsGuardFailure(e);
+            return false;
+        }
+    }
+
+    private handleSpecificAnswerToAskedQuestion(input: ControlInput, resultBuilder: ControlResultBuilder) {
+        const content = this.getQuestionnaireContent(input);
+        const question = this.getQuestionContentById(this.state.focusQuestionId!, input);
+
+        const { feedback, action, target, valueStr, valueType, erMatch } = unpackSingleValueControlIntent(
+            (input.request as IntentRequest).intent,
+        );
+
+        // Accept the answer if it is a match to one of the listed choices, otherwise
+        // report error and ignore.  Note: being a match is a stronger condition than
+        // simply being an ER_MATCH as the answer slot type might have more values than
+        // answers for a particular questionnaire.
+
+        if (content.choices.some((choice) => choice.id === valueStr)) {
+            const choiceIndex = this.getChoiceIndexById(content, valueStr);
+            assert(choiceIndex !== undefined);
+            const questionIndex = this.getQuestionIndexById(content, this.state.focusQuestionId!);
+
+            this.updateAnswer(question.id, valueStr, input, resultBuilder);
+
+            resultBuilder.addAct(
+                new QuestionAnsweredAct(this, {
+                    questionId: question.id,
+                    choiceId: valueStr,
+                    userAnsweredWithExplicitValue: true,
+                    userMentionedQuestion: false,
+                    renderedChoice: content.choices[choiceIndex].prompt,
+                    renderedQuestionShortForm: this.evaluatePromptShortForm(
+                        content.questions[questionIndex].promptShortForm,
+                        input,
+                    ),
+                }),
+            );
+        } else {
+            throw new Error('todo');
```

Comment:
I think the else case is to handle invalid values spoken by voice, does not apply for APL touch

Reply:
no longer present.  (there is still the one for voice)